lib/screens/setup/device_setup_screen.dart

```diff
@@ -47,28 +52,29 @@ class SyncConfigScreen extends StatelessWidget {
             },
           ),
           Center(
-            child: TextButton.icon(
-              icon: Icon(Icons.add),
-              onPressed: () => Navigator.of(context).push(
-                MaterialPageRoute(
-                  builder: (_) =>
-                      QrScannerScreen(onQRCodeRead: viewModel.handleQrCodeRead),
-                ),
-              ),
-              onLongPress: () => Navigator.of(context).push(
-                MaterialPageRoute(
-                  builder: (_) => DeviceFormScreen(),
+            child: Column(
+              children: [
+                if (viewModel.showQrScannerButton)
+                  TextButton.icon(
+                    icon: Icon(Icons.qr_code_scanner_rounded),
+                    onPressed: () => _showQRScannerScreen(context, viewModel),
+                    label: Text('Scan QR code'),
+                  ),
+                if (viewModel.showQrScannerButton) Text('or'),
```

Comment:
It looks a bit weird to have the same if condition in two consecutive statements.

Maybe do something like this:
```dart
if (viewModel.showQrScannerButton) ...[
  TextButton.icon(
    icon: Icon(Icons.qr_code_scanner_rounded),
    onPressed: () => _showQRScannerScreen(context, viewModel),
    label: Text('Scan QR code'),
  ),
  Text('or'),
],
```
Not totally sure if this is more readable though...